### PR TITLE
Feat: follow unfollow

### PR DESCRIPTION
**Pull request by M-Byte480**
Add a user to your DB with id 100 per say and test user.

Start up Gateway, Service Registry, Auth, and Profile microservices

Login through Google Oauth 2.0, get your JWT token and add it to postman. 

GET http://localhost:8080/profile/test to see if the API is running. if you get 503 give it a few seconds until Eureka has discovered it

POST http://localhost:8080/profile/follow/100
Where 100 is the UserId you want to follow. THe response should be your newly cached follower following count. You can keep pinging this to unfollow. The DB will check the history of you interacting with the user

### CODE REVIEW

**Review by Silver-Golden — APPROVED** [review]
Looks good to me

**Comment by M-Byte480** [comment]
Rebased branch

File: Microservices/Posts/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Stage 1: Build
+FROM maven:3.9.9-eclipse-temurin-21 AS builder
+WORKDIR /app
+COPY pom.xml .
+COPY src ./src
+RUN mvn clean package -DskipTests
+
+# Stage 2: Run
+FROM openjdk:21-jdk-slim
```
Comment (Bourkupine) [inline Microservices/Posts/Dockerfile:9]: we are using temurin distributor elsewhere in the project, will this cause any problems?

File: docker-compose.yml [hunk 2]
```diff
@@ -49,46 +68,67 @@ services:
...
+        # name of teh folder in Microservices
+        SERVICE_PATH: "Profile"
+    # pass in the env file to be used
     env_file: ".env"
     environment:
+      PROFILE_DB_IP: profile_mysql
       SERVICE_REGISTRY_IP: eureka-service
+    ports:
+      # ideally it would be mapped to a more static internal port
+      # - "${PROFILE_MICROSERVICE_PORT}:8080"
```
Comment (Bourkupine) [inline docker-compose.yml:92]: why not do this? if theres a solid reason for not then id say remove these comments

File: Microservices/Profile/src/main/java/cs4337/group_8/ProfileService/controllers/ControllerProfile.java [hunk 3]
```diff
@@ -25,53 +22,37 @@ public class ControllerProfile {
...
-
-    public ControllerProfile(ProfileService profileService) {
+    private final JwtService jwtService;
+    public ControllerProfile(ProfileService profileService,
+                             JwtService jwtService) {
         this.profileService = profileService;
+        this.jwtService = jwtService;
     }
 
     @GetMapping("/test")
```
Comment (Bourkupine) [inline Microservices/Profile/src/main/java/cs4337/group_8/ProfileService/controllers/ControllerProfile.java:32]: do we need to keep this in?

**Review by Bourkupine — COMMENTED** [review]
Looks good, just left a few comments (all minor nits/questions).

File: Microservices/Profile/src/main/java/cs4337/group_8/ProfileService/controllers/ControllerProfile.java (hunk 3 above)
Comment (M-Byte480) [reply Microservices/Profile/src/main/java/cs4337/group_8/ProfileService/controllers/ControllerProfile.java:32]: It's Healthcheck

File: Microservices/Posts/Dockerfile (hunk 1 above)
Comment (M-Byte480) [reply Microservices/Posts/Dockerfile:9]: Only profile has the different one. All the other ones use the openjdk

**Review by killianC2323 — APPROVED** [review]
All looks good to me

**Pull request merged**